geps/gep-1651.md

```diff
@@ -0,0 +1,268 @@
+>
+>   IPAddress
+>   IPv4Address
+>   IPv6Address
+>   ClusterLocalIPAddress
+>   ClusterLocalIPv4Address
+>   ClusterLocalIPv6Address
+>
+> For each dimension we avoid adding a separate field for and instead try to embed into a single name, we risk this kind of name explosion. Of course none of the above even begins to cover my idea of NetworkLocal which could further complicate this.
+
+### Scope/reachability field on GatewayAddress
+
+This would allow Gateways to have multiple scopes.
+
+From: https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1486271913
+> The obvious application for multiple scopes seems to be saving on boilerplate, which is a win, but are there are any other advantages to allowing one Gateway to have multiple scopes?
+>
+> Multiple scopes Pros:
+>
+> Allows a single Gateway to express multiple networks, saving on needing to attach HTTPRoutes to multiple Gateways for each network scope.
+>
+> Multiple scopes Cons:
+>
+> Complicates the Gateway's purpose. Instead of one Gateway being one set of Listeners, now a Gateway is two sets of listeners that have a totally different scope (and presumably, security context). Personally, I'm also concerned how this will interact with other features like merging and preprovisioning that GEP-1867: Per-Gateway Infrastructure #1868 will allow.
+
+### Adding `reachabiliy` attribute to GatewayClass
+
+See [Prior Art - Multiple Gateways Classes](#multiple-gateway-classes)
+
+## Survey of Prior Art
+
+These alternatives are a survey of existing approaches to support cluster
+local Gateways. Most are implementation specific and are not portable.
+
+### Special annotation/label
+
+Istio let's you specify an annotation `networking.istio.io/service-type` to
+change the underlying Kubernetes Service type to make it a ClusterIP type.
+
+### Re-use of AddressType Hostname
+
+Istio let's you re-use existing Gateway deployments by setting the address
+type to `Hostname` and the value to the Istio ingress Kubernetes Service. If an
+operator configures the Istio deployment to support cluster local traffic a
+Gateway implementation can select it using the `HostName` attribute.
+
+### Multiple Gateway Classes
+
+Some implementations support multiple deployments on a single cluster where each maps to a
+GatewayClass. One of these deployments can be configured to serve cluster local traffic. This is
+sub-optimal because this is implementation specific and the end-user is effectively managing the
+deployments themselves rather than infrastructure being automatically provisioned.
+
+Likewise, infrastructure providers may provide a fixed set of GatewayClasses with unique and fixed
+reachability. Thus GatewayClass name is a viable option to control reachability. But it is not
+flexible enough for implementations that can configure reachability when creating Gateways.
```

Comment:
> Wouldn't it be true that any Gateway implementation that supported multiple forms of reachability could support that config on GatewayClass or Gateway?

Maybe - I really can't speak to that.

> If so, this line doesn't really make sense to me.


My comment [here](https://github.com/kubernetes-sigs/gateway-api/pull/1653/files/28cefa9f7a4e1f5e7571bc5815d5b0e861929a2c#r1162203168) highlights why I included this paragraph

> This is theoretical for me as I'm not familar with how different implementations map infrastructure to the Gateway types.
>
> My point here is if there is overhead when creating a GatewayClass then suggesting we create more classes is less efficient than just having a field on the Gateway toggling reachability.
>
> ie. if there is no overhead infrastructure then you're still dealing with the etcd storage of the having GatewayClasses when it might be unnecessary.

Do you have a suggestion on my wording here?

Reply:
Changed to wording to say it's not as flexible for _end-users_ who may want to toggle reachability at the gateway level